Library/Homebrew/utils/topological_hash.rb

```diff
@@ -1,16 +1,23 @@
-# typed: true # rubocop:todo Sorbet/StrictSigil
+# typed: strict
 # frozen_string_literal: true
 
 require "tsort"
 
 module Utils
   # Topologically sortable hash map.
   class TopologicalHash < Hash
+    extend T::Generic
     include TSort
 
+    CaskOrFormula = T.type_alias { T.any(Cask::Cask, Formula) }
+
+    K = type_member { { fixed: CaskOrFormula } }
+    V = type_member { { fixed: T::Array[CaskOrFormula] } }
+    Elem = type_member(:out) { { fixed: [CaskOrFormula, T::Array[CaskOrFormula]] } }
```

Comment:
These lines feel redundant, but i couldn't convince sorbet to accept anything more concise.